addons/medical_feedback/functions/fnc_playInjuredSound.sqf

```diff
@@ -26,60 +26,35 @@ params [["_unit", objNull, [objNull]], ["_type", "hit", [""]], ["_severity", 0,
 if (!local _unit) exitWith {
     ERROR("Unit not local or null");
 };
-
 if !(_unit call EFUNC(common,isAwake)) exitWith {};
 
-private _timeOut = [TIME_OUT_HIT, TIME_OUT_MOAN] select (_type == "moan");
-
 // Handle timeout
+private _timeOut = [TIME_OUT_HIT, TIME_OUT_MOAN] select (_type == "moan");
 if (_unit getVariable [QGVAR(soundTimeout) + _type, -1] > CBA_missionTime) exitWith {};
 _unit setVariable [QGVAR(soundTimeout) + _type, CBA_missionTime + _timeOut];
 
-// Get sounds
-private _soundsNamespace = NAMESPACE_NULL;
-
-switch (toLower _type) do {
-    case ("hit"): {
-        _soundsNamespace = GVAR(HitScreamNamespace);
-    };
-    case ("moan"): {
-        _soundsNamespace = GVAR(InjuredMoanNamespace);
-    };
-};
-
 // Get units speaker
 private _speaker = speaker _unit;
-
 if (_speaker == "ACE_NoVoice") then {
     _speaker = _unit getVariable "ace_originalSpeaker";
 };
 
-private _sounds = _soundsNamespace getVariable _speaker;
-
-if (isNil "_sounds") then {
-    _sounds = _soundsNamespace getVariable (_soundsNamespace getVariable "#default");
-};
-
-if (isNil "_sounds") exitWith {
-    ERROR("No sounds for speaker and no default found");
-};
-
-// Get correct sound of the speaker
-_sounds = _sounds param [_severity, []];
-(selectRandom _sounds) params ["_sound", ["_volume", 1], ["_frequency", 1], ["_distance", 80]];
-
-if (isNil "_sound") exitWith {
-    ERROR("No sound for this speaker");
-};
-
-// Delete leading slash.
-if (_sound select [0, 1] == "\") then {
-    _sound = _sound select [1];
+// Fallback if speaker has no associated scream/moan sound
+if (isNull (configFile >> "CfgSounds" >> format ["ACE_moan_%1_low_1"])) then {
+    _speaker = "Male08ENG";
 };
 
-// Default file extension.
-if (_sound find "." == -1) then {
-    _sound = _sound + ".wss";
+// Select actual sound
+private _variation = ["low", "mid", "high"] select _severity;
+private _distance = if (_type == "hit") then {
+    [50, 60, 70] select _severity;
+} else {
+    [10, 15, 20] select _severity;
 };
+private _filter = format ["(configName _x) find 'ACE_%1_%2_%3' == 0", _type, _speaker, _variation];
+private _sounds = _filter configClasses (configFile >> "CfgSounds");
+private _sound = configName selectRandom _sounds;
 
-playSound3D [_sound, objNull, false, position _unit, _volume, _frequency, _distance];
+// Limit network traffic by only sending the event to players who can potentially hear it
+private _targets = (ASLToAGL getPosASL _unit) nearEntities ["CAManBase", _distance];
```

Comment:
nearEntities should be able to take object on left hand side

Reply:
I don't think we should filter for players. If it's not a player, then it will likely be a HC or server owned unit, and the CBA variable will be sent to the server either way. Extra traffic to the HC doesn't hurt because it should run on the same machine as the server.